capture/split.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import gzip
+from Bio import SeqIO
+
+
+def __split__(args):
+    logger = logging.getLogger(__name__)
+    for file in args.input_file.split(","):
+        wanted_cov = 100
+        # caclul the number of reads in each subsample for a coverage
+        number_records = (wanted_cov*args.genome-size)/args.length_read
+        if file.endswith(".bam", ".sam"):
```

Comment:
As said on call, you could try to use magic numbers to identify bam and gzipped files.
the SAM/BAM format spec is here: [https://samtools.github.io/hts-specs/SAMv1.pdf](https://samtools.github.io/hts-specs/SAMv1.pdf)